### PR TITLE
fix: only throw MissingCRDException if we get a 404 on the target CRD

### LINKED ISSUE #552: Misleading MissingCRDException exception
If an event source fails to start and the error code is 404, the `DefaultEventSourceManager` wraps it with a `MissingCRDException` [1] which the `ConfiguredController` re-wrap and include details about the CRD of the controller resulting in a misleading message as the exception will report that the CRD of the controller is missing but it may not be the case.

[1] https://github.com/java-operator-sdk/java-operator-sdk/blob/master/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/DefaultEventSourceManager.java#L92-L97
[2] https://github.com/java-operator-sdk/java-operator-sdk/blob/master/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/ConfiguredController.java#L177-L182

### PR DESCRIPTION
**Pull request by metacosm**
Note that this relies on fabric8io/kubernetes-client#3492 so won't build
until this PR is merged and available in a release.
Fixes #552

### CODE REVIEW

**Comment by lburgazzoli** [comment]
LGTM

wonder if the MissingCRDException should only be thrown by the CustomResourceEventHanderl instead of any event source

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/DefaultEventSourceManager.java [hunk 1]
```diff
@@ -92,7 +95,10 @@ public final void registerEventSource(String name, EventSource eventSource)
       if (e instanceof KubernetesClientException) {
```
Comment (csviri) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/DefaultEventSourceManager.java:95]: Should not we catch this exception and make these checks in the CustomResourceEventSource? 
(And just throw the exception further if this is not the exception we want to trasnlate)

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/DefaultEventSourceManager.java [hunk 2]
```diff
@@ -92,7 +95,10 @@ public final void registerEventSource(String name, EventSource eventSource)
...
         KubernetesClientException ke = (KubernetesClientException) e;
         if (404 == ke.getCode()) {
```
Comment (csviri) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/DefaultEventSourceManager.java:97]: I mean this logic. This is quite specific to CustomResourceEventSource. Cannot be moved there?

**Review by csviri — APPROVED** [review]
LGTM!

Nice late night PR :)

**Pull request merged**